src/Simple.OData.Client.Core/Fluent/FluentCommand.cs

```diff
@@ -188,6 +188,15 @@ public FluentCommand WithMedia(params ODataExpression[] properties)
             return WithMedia(properties.Select(x => x.Reference));
         }
 
+        public FluentCommand WithHeaders(IDictionary<string, string> operationHeaders)
+        {
+            if (IsBatchResponse)
```

Comment:
This should include a method level comment that headers are not applied per-request in a batch response.

Reply:
Will do